modules/tensor_mechanics/include/postprocessors/MaterialTimeStepPostprocessor.h

```diff
@@ -31,9 +31,33 @@ class MaterialTimeStepPostprocessor : public ElementPostprocessor
   virtual void threadJoin(const UserObject & y);
 
 protected:
-  const MaterialProperty<Real> & _matl_time_step;
+  /// Flag to find the time step limit from material properties
+  const bool _use_matl_time_step;
 
-  Real _value;
+  /// Pointer to the material property containing the time step limit
+  const MaterialProperty<Real> * _matl_time_step;
+
+  /// Current time step limit from the material properties
+  Real _matl_value;
+
+  /// Flag to limit the time step based on the number of elements changed
+  const bool _use_elements_changed;
+
+  ///@{ Material property used to determine if elements have changed
+  const MaterialProperty<Real> * _changed_property;
+  const MaterialProperty<Real> * _changed_property_old;
+  ///@}
+
+  /// Target number of changed elements used to determine if we need to change the time step
+  const int _target;
```

Comment:
Some of these variables could be re-named to make them a little more clear.  Some suggestions:

_target->_elements_changed
_tolerance->_elements_changed_threshold